include/deal.II/lac/sparse_direct.h

```diff
@@ -427,6 +432,119 @@ class SparseDirectUMFPACK : public EnableObserverPointer
   std::vector<double> control;
 };
 
+
+/**
+ * This class provides an interface to the parallel sparse direct solver <a
+ * href="https://mumps-solver.org">MUMPS</a>. MUMPS is direct method based on
+ * a multifrontal approach, which performs a direct LU factorization. The
+ * matrix coming in may have either symmetric or nonsymmetric sparsity
+ * pattern.
+ *
+ * @note This class is useable if and only if a working installation of <a
+ * href="https://mumps-solver.org">MUMPS</a> exists on your system and was
+ * detected during configuration of <code>deal.II</code>.
+ *
+ * <h4>Instantiations</h4>
+ *
+ * There are instantiations of this class for SparseMatrix<double>,
+ * SparseMatrix<float>, BlockSparseMatrix<double>, and
+ * BlockSparseMatrix<float>.
+ */
+class SparseDirectMUMPS : public EnableObserverPointer
+{
+private:
+#ifdef DEAL_II_WITH_MUMPS
+  DMUMPS_STRUC_C id;
+#endif // DEAL_II_WITH_MUMPS
+
+  double                 *a;
+  std::vector<double>     rhs;
+  int                    *irn;
+  int                    *jcn;
```

Comment:
I think we need to convert these plain pointers to containers that automatically manage memory, e.g. `std::unique_ptr<double[]>`.